### PR TITLE
Cleanup for Stackdriver config

### PR DESCRIPTION
**Pull request by jkohen**
Proposing changes to the documentation suggested in external review.

We also agreed to remove SinkInfo. Would you like me to do that in this PR?

### CODE REVIEW

File: mixer/adapter/stackdriver/config/config.proto [hunk 1]
```diff
@@ -40,223 +40,177 @@ option (gogoproto.gostring_all) = false;
...
-    // Endpoint URL to send stackdriver data - leave empty to use the
-    // StackDriver SDK's default value (monitoring.googleapis.com).
-    string endpoint = 1;
-
-    // GCP Project to attach metrics to.
-    string project_id = 2;
-
-    // This adapter batches the data it sends to Stackdriver; we will push to stackdriver every push_interval.
-    // If no value is provided we default to once per minute.
-    google.protobuf.Duration push_interval = 3 [(gogoproto.nullable)=false, (gogoproto.stdduration) = true];
```
Comment (douglas-reid) [inline mixer/adapter/stackdriver/config/config.proto]: Why are we eschewing this config?  Do we believe that this was not being customized at all?

File: mixer/adapter/stackdriver/config/config.proto [hunk 2]
```diff
@@ -40,223 +40,177 @@ option (gogoproto.gostring_all) = false;
...
-
-    // Credentials must be attached to every call made to StackDriver. This
-    // can be an API key, a service account secret key, or Oauth2 tokens.
-    //
-    // TODO: do we want to support oauth?
-    oneof creds {
```
Comment (douglas-reid) [inline mixer/adapter/stackdriver/config/config.proto]: By removing this, are we insisting then that SD adapters only be configured on GCE/GKE installs?  How do non-GCE/GKE handlers attach service acct creds without such a place to hang them?

File: mixer/adapter/stackdriver/config/config.proto [hunk 3]
```diff
@@ -40,223 +40,177 @@ option (gogoproto.gostring_all) = false;
...
+    google.api.MetricDescriptor.MetricKind kind = 1;
+    // The type of the metric's value. Ex: Distribution.
+    google.api.MetricDescriptor.ValueType value = 2;
+
+    // For metrics with a metric value of DISTRIBUTION, this provides a
+    // mechanism for configuring the buckets that will be used to store the
+    // aggregated values. This field must be provided for metrics declared to
+    // be of type DISTRIBUTION. This field will be ignored for
+    // non-distribution metric kinds.
+    google.api.Distribution buckets = 3;
```
Comment (douglas-reid) [inline mixer/adapter/stackdriver/config/config.proto]: Isn't this a breaking change for existing configuration?  Should we have a `google.api.Distribution distribution_buckets = 5` ?

File: mixer/adapter/stackdriver/config/config.proto [hunk 4]
```diff
@@ -40,223 +40,177 @@ option (gogoproto.gostring_all) = false;
...
+    // If an HttpRequestMapping is provided, a HttpRequest object will be
+    // filled out for this log entry using the variables named in the mapping
+    // to populate the fields of the request struct from the instance's
+    // variables.
+    HttpRequestMapping http_mapping = 3;
+  }
+
+  // Describes how to represent an Istio Log in Stackdriver.
+  // Describes how to represent an Istio Log in Stackdriver.
+  message LogInfo {
```
Comment (douglas-reid) [inline mixer/adapter/stackdriver/config/config.proto]: this is here twice? cut/paste issue?  or is my diff view glitching?

File: mixer/adapter/stackdriver/config/config.proto [hunk 5]
```diff
@@ -40,223 +40,177 @@ option (gogoproto.gostring_all) = false;
...
-    // Stackdriver Trace configuration.
-    Trace trace = 10;
+    // If SinkInfo is provided, Stackriver logs would be exported to that sink.
+    SinkInfo sink_info = 4;
+  }
+  // A map of Istio LogEntry name to Stackdriver log info.
+  map<string, LogInfo> log_info = 9;
+
+  // Details of Stackdriver Trace configuration for tracespan template.
+  message Trace {
```
Comment (douglas-reid) [inline mixer/adapter/stackdriver/config/config.proto]: I think we might want to think about deprecating this too, moving forward.

**Comment by jkohen** [comment]
As discussed with Doug, I'm going to drop this change in favor of a Stackdriver-specific template.

**Pull request closed without merging**